include/dr/shp.hpp

```diff
@@ -4,10 +4,6 @@
 
 #pragma once
 
-#include <dr/detail/iterator_adaptor.hpp>
```

Comment:
@BenBrock , please check this change. If something more should be removed or something was wrongly removed, then please write. Files here should include only direct API used by external users (no details), see this thread: https://github.com/oneapi-src/distributed-ranges/pull/190#pullrequestreview-1391885986

Reply:
I probably added that before we agreed on current approach.

Reply:
I will revert this change if we will not agree on shrinking mhp.hpp / shp.hpp by removing non-public headers.